python/ccxt/base/exchange.py

```diff
@@ -1497,7 +1497,7 @@ def build_ohlcvc(self, trades, timeframe='1m', since=None, limit=None):
         ohlcvs = []
         (timestamp, open, high, low, close, volume, count) = (0, 1, 2, 3, 4, 5, 6)
         num_trades = len(trades)
-        oldest = (num_trades - 1) if limit is None else min(num_trades - 1, limit)
+        oldest = num_trades if limit is None else min(num_trades, limit + 1)
```

Comment:
@auchri question why adding +1 to the limit here `min(num_trades, limit + 1)` ?

Reply:
@carlosmiei Because then it is consistent with the javascript function (equal number of trades are used then)

Reply:
@auchri afaik the JS version does this: ` Math.min (trades.length - 1, limit);` but the goal is only to avoid out-of-range access, for instance array[3] when array.length is 3, not sure if adding 1 to the limit is equivalent

Reply:
Yes, but the javascript loop also uses the highest item: `for (let i = 0; i <= oldest; i++)` while the python loop is excluding the highest item

So with my change out of range errors are still avoided and the limit parameter is equally handled as in javascript

Reply:
> Yes, but the javascript loop also uses the highest item: `for (let i = 0; i <= oldest; i++)` while the python loop is excluding the highest item

@auchri I agree with this, I'm just not convinced yet why we need to add 1 to the limit (sorry I'm missing something)

Let's say you have 10 trades and your limit is 5, why would the oldest trade be the 6th? the loop below would iterate from 0 to 5 `  for i in range(0,6)`  and catch 6 trades instead of 5

Reply:
@carlosmiei I get your point, but at least this way it is at least consistent with javascript:
<img width="797" alt="image" src="https://user-images.githubusercontent.com/5092164/236294813-8eac6710-6470-4906-9ea3-67eb1b58ba55.png">

<img width="710" alt="image" src="https://user-images.githubusercontent.com/5092164/236294918-44010b0d-d15f-4354-9ce5-3a32716e386e.png">

Maybe it should be fixed in javascript that if the limit is 5 then only 5 trades are used. Should I create a PR and modify this one?

Reply:
> @carlosmiei I get your point, but this way it is at least consistent with javascript: <img alt="image" width="797" src="https://user-images.githubusercontent.com/5092164/236294813-8eac6710-6470-4906-9ea3-67eb1b58ba55.png">
> 
> <img alt="image" width="710" src="https://user-images.githubusercontent.com/5092164/236294918-44010b0d-d15f-4354-9ce5-3a32716e386e.png">
> 
> Maybe it should be fixed in javascript that if the limit is 5 then only 5 trades are used. Should I create a PR and modify this one?

@auchri Yeah I think so, but you don't need to create another PR, can you compile all changes to buildohlcv in this one? it's easier to compare the different implementations